Sorting/BubbleSort.swift

```diff
@@ -0,0 +1,50 @@
+//
+//  BubbleSort.swift
+//  Sorting
+//
+//  Created by Michael Woodruff on 02/11/2016.
+//  Copyright © 2016 Mike Woodruff. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+class BubbleSort: SortingAlgorithm {
+    
+    typealias MoveType = BubbleSortMove
+    static var name = NSLocalizedString("bubble_sort_title", comment: "");
+    
+    static var worstComplexity = BigOhStringFormatter.bigOhOfNSquared();
+    static var averageComplexity = BigOhStringFormatter.bigOhOfNSquared();
+    static var bestComplexity = BigOhStringFormatter.bigOofN();
+    static var worstCase: [Int] = InputArrays.randomSortedArray(length: 5, ascending: false);
+    static var bestCase:[Int] = InputArrays.randomSortedArray(length: 5, ascending: true);
+    
+    static func sort(unsortedArray: [Int]) -> [MoveType] {
+     
+        var sortedAboveIndex = unsortedArray.count;
+        var sortedArray = unsortedArray;
+        var moves: [BubbleSortMove] = [];
+        
+        repeat {
+            var lastSwapIndex = 0;
+            for i in 1 ..< sortedAboveIndex {
+                let firstPosition: Position = Position(index: i, value: sortedArray[i]);
+                let secondPosition: Position = Position(index: i-1, value: sortedArray[i-1]);
```

Comment:
Can it be using one variable storing i-1? Because it will be used many times as following logic